test/services/IdentityMatchServiceSpec.scala

```diff
@@ -16,60 +16,96 @@
 
 package services
 
-import connectors.IdentityMatchConnector
+import com.github.tomakehurst.wiremock.client.WireMock.{aResponse, post, urlEqualTo}
 import exceptions.LimitException
-import models.IdMatchResponse
 import models.api1585.{IdMatchApiError, NinoNotFound}
+import models.{IdMatchResponse, OperationSucceeded}
 import org.mockito.ArgumentMatchers.{any, eq => mockEq}
-import org.mockito.Mockito.{times, verify}
-import play.api.Application
-import play.api.inject.bind
+import org.mockito.Mockito.{reset, times, verify, when}
+import play.api.http.Status.{NOT_FOUND, OK}
+import play.api.libs.json.{JsValue, Json}
 import play.api.test.{DefaultAwaitTimeout, FutureAwaits}
-import repositories.IndividualCheckRepository
-import uk.gov.hmrc.http.HttpClient
 import util.{BaseSpec, IdentityMatchHelper}
 
-import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
-class IdentityMatchServiceSpec extends BaseSpec with IdentityMatchHelper with FutureAwaits with DefaultAwaitTimeout {
+class IdentityMatchServiceSpec extends BaseSpec
+  with IdentityMatchHelper
+  with FutureAwaits
+  with DefaultAwaitTimeout {
 
-  val identityMatchConnector: IdentityMatchConnector = application.injector.instanceOf[IdentityMatchConnector]
-  val identityMatchService: IdentityMatchService = application.injector.instanceOf[IdentityMatchService]
+  val individualsMatchUrl = "/individuals/match"
 
-  override lazy val application: Application = applicationBuilder()
-    .overrides(
-      bind[HttpClient].toInstance(httpClient),
-      bind[IndividualCheckRepository].toInstance(mockIndividualCheckRepository),
-      bind[AuditService].toInstance(mockAuditService)
-    ).build()
+  lazy val identityMatchService: IdentityMatchService = application.injector.instanceOf[IdentityMatchService]
+
+  override def beforeEach(): Unit = {
+    reset(mockAuditService)
+    reset(mockIndividualCheckRepository)
+
+    when(mockIndividualCheckRepository.incrementCounter(any())).thenReturn(Future.successful(OperationSucceeded))
+    when(mockIndividualCheckRepository.getCounter(any())).thenReturn(Future.successful(0))
+    when(mockIndividualCheckRepository.clearCounter(any())).thenReturn(Future.successful(OperationSucceeded))
+  }
 
   "Identity Match Connector" should {
 
     "parse response correctly" when {
 
       "success is returned" in {
+
+        wireMockServer.stubFor(
```

Comment:
Create method for this stub:
```scala annotate
  def createMockForIndividualMatchUrl(returnStatus: Int, errorResponse: JsValue, individualsMatchUrl: String = "/individuals/match"): StubMapping =
    wireMockServer.stubFor(
      post(urlEqualTo(individualsMatchUrl)).willReturn(
        aResponse()
          .withStatus(returnStatus)
          .withBody(errorResponse.toString())
      )
    )
```
can save some lines in each test

Reply:
This can be put in the `BaseSpec` trait and used in the `IndividualCheckControllerSpec` as well.

Reply:
Added method thank you